### DISCUSSION TITLE
Where is Wix.targets?

### DISCUSSION THREAD

**Discussion by dlidstrom** [community/community#26747] [Actions]
<p>I had a setup where I could use Wix through msbuild, something which requires importing Wix.targets. However it has stopped working. I am using windows-2019 runner and I’m seeing this error:</p>
<pre><code class="lang-auto">
D:\a\Snittlistan\Snittlistan\build.build(16,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\WiX\v3.x\Wix.targets" was not found. Confirm that the expression in the Import declaration "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\WiX\v3.x\Wix.targets" is correct, and that the file exists on disk.
</code></pre>
<p>The relevant part of my msbuild script is this:</p>
<pre><code class="lang-auto"> &lt;PropertyGroup&gt;
    &lt;WixTargetsPath Condition=" '$(WixTargetsPath)' == '' "&gt;$(MSBuildExtensionsPath)\Microsoft\WiX\v3.x\Wix.targets&lt;/WixTargetsPath&gt;
…
</code></pre>
<p>Where has the file gone?</p>

**Comment by AF250329** [comment]
<p>Oooohhh… I thought it happens only to me… yes - I can confirm wix.targets files is gone… :frowning:</p>
<p>Actual error that I get:</p>
<pre><code class="lang-auto">error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\WiX\v3.x\Wix.targets" was not found. Confirm that the expression in the Import declaration "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\WiX\v3.x\Wix.targets" is correct, and that the file exists on disk.
</code></pre>
<p>It seems that  Windows runner does not install Wix Extension any more into Visual Studio :frowning:</p>
<p>Trying to test suggestion from here: <a href="https://github.com/wixtoolset/issues/issues/5804#issuecomment-532663590" class="inline-onebox" rel="noopener nofollow ugc">WiX toolset does not install targets into Microsoft Build Tools 2017 folder · Issue #5804 · wixtoolset/issues · GitHub</a></p>

**Comment by dlidstrom** [comment]
<p>Are you able to apply the suggestion?</p>

**Comment by dlidstrom** [comment, marked as answer]
<p>This solved my issue:</p>
<aside class="onebox githubissue" data-onebox-src="https://github.com/actions/virtual-environments/issues/4419#issuecomment-978916739">
  <header class="source">

      <a href="https://github.com/actions/virtual-environments/issues/4419#issuecomment-978916739" target="_blank" rel="noopener nofollow ugc">github.com/actions/virtual-environments</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/actions/virtual-environments/issues/4419#issuecomment-978916739" target="_blank" rel="noopener nofollow ugc">Add WiX Toolset build tools to `windows-2022`</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-11-03" data-time="16:57:14" data-timezone="UTC">04:57PM - 03 Nov 21 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/barnson" target="_blank" rel="noopener nofollow ugc">
          <img alt="barnson" src="https://user-images.githubusercontent.com/371713/181099733-3bceddef-0002-4955-9899-3d46b3a42b83.jpeg" class="onebox-avatar-inline" width="20" height="20">
          barnson
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          awaiting-deployment
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          OS: Windows
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          community-feature
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          feature request
        </span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">### Tool name

WiX Toolset

### Tool license

MS-RL

### Add or update?

- [X] A<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">dd
- [ ] Update

### Desired version

v3.11.2.4516

### Approximate size

120MB

### Brief description of tool

About the WiX toolset: The WiX toolset lets developers create installers for Windows Installer, the Windows installation engine.

WiX is available in previous generations of VMs but was not included in `windows-2022`, presumably because Visual Studio 2022 isn't yet supported by the WiX Visual Studio extension. However, the WiX Visual Studio extension is only used for in-IDE development and isn't needed to build (from the command line or MSBuild).

### URL for tool's homepage

https://wixtoolset.org/

### Provide a basic test case to validate the tool's functionality.

```bash
"%WIX%\bin\candle.exe"
```


### Virtual environments affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Can this tool be installed during the build?

_No response_

### Tool installation time in runtime

_No response_

### Are you willing to submit a PR?

Yes</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
